server/utils/premises/occupancy.ts

```diff
@@ -97,19 +99,66 @@ export const generateDaySummaryText = (daySummary: Cas1PremisesDaySummary): stri
   return messages.length ? `This AP ${messages.join(' and ')}.` : ''
 }
 
-export const daySummaryRows = (daySummary: Cas1PremisesDaySummary) => {
+const availabilityRow = (
+  name: string,
+  characteristic: Cas1SpaceBookingCharacteristic,
+  available: number,
+  booked: number,
+) => {
+  return booked || available
+    ? {
+        key: { text: name },
+        value: {
+          html: `${pluralize('bed', available)}${booked ? `<span class="govuk-!-padding-right-2"></span><a href="${characteristic ? `?characteristics=${characteristic}` : '?'}"> ${pluralize('booking', booked)}</a>` : ''} ${(booked || available) && booked >= available ? `<strong class="govuk-tag govuk-tag--${booked > available ? 'red' : 'yellow'} govuk-tag--float-right">${booked > available ? 'Overbooked' : 'Full'}</strong>` : ''}`,
+        },
+      }
+    : null
+}
+
+export const daySummaryRows = (
+  daySummary: Cas1PremisesDaySummary,
+  roomCharacteristics: Array<Cas1SpaceBookingCharacteristic> = null,
+  characteristicsMode: 'singleRow' | 'doubleRow' | 'none' = 'none',
```

Comment:
I was wondering why I was seeing code to do with tags and such but not seeing them in the browser... Nice!

This should _probably_ be in a separate branch, but I'll let it live here 🙂

Reply:
Yes, I jumped the gun a bit putting this up for review. I'll put a feature flag round it to turn it on.